### PR TITLE
docs: Draft breaking changes doc for contributors

### LINKED ISSUE #56: DOC: Breaking Changes Guide
There is a request to provide information about breaking changes for the purposes of educating partners and customers.

Follow up with Tobias and Farzin for more info about the scope of this guide.

### PR DESCRIPTION
**Pull request by marlass**
This covers guideline for contributors.
New doc should also be created about handling those breaking changes on version updates, general deprecation policy, making @alpha and @beta API access, automatic migrations, etc.

Those things need to be discussed first. After that, we can provide docs for those topics.

Closes #56

### CODE REVIEW

File: _pages/contributing/breaking-changes.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+title: Breaking changes (DRAFT)
+---
+
+## Avoiding breaking changes
+
+Below is a list of things should avoid doing while working on new features and bug fixes. We try to following semantic versioning and doing that means keeping breaking changes for major releases. This list should make you aware of different ways to introduce breaking changes.
+
+- don't change selector of component/directive/pipe
+- don't change or remove exported classes, functions, constants, interfaces and types in `public_api.ts` file
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md]: (note: index.ts barrels also can bubble up the exports to public_api.ts)

File: _pages/contributing/breaking-changes.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- don't change or remove interface properties
+- don't change or remove public/protected class members
+- don't change or remove existing public functions parameters (also don't change order)
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md]: public/protected

File: _pages/contributing/breaking-changes.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- don't add new required properties to interfaces and types (only optional with '?')
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md]: For ease of reading, can we move this line after all function-related ones (so function-related are grouped)?

File: _pages/contributing/breaking-changes.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+- don't add new required parameters to public functions (only optional)
+- don't change function behavior (after change it should return the same output for provided input and dispatch same side effects). Bug fixes might require different function behavior for edge cases.
+- don't change, remove unit tests (you can only change internal implementation details, assertions shouldn't be changed)
+- don't remove anything from Angular modules (imports, providers, declarations, exports)
+- check generated public API report with previous `develop` report. Make sure only things that you explicitly wanted exported are visible publicly.
+- don't remove or update existing values for public enums (only add new values)
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md]: - for *exported* enums
- *or change order*

reason: enums that don't have initial values are implicitly numbered, so changing order may change the value
```
enum MyEnum {
  foo, // implicit 0
  bar // implicit 1
}
````

File: _pages/contributing/breaking-changes.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+- default value for new functions parameters should match function behavior before change
+- any change to class constructor is breaking (`super` calls needs to be updated in classes extending ours)
+- don't change exported class fields to readonly
+- don't change or remove translations
+
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md]: Please add: - don't remove or change default configs

File: _pages/contributing/breaking-changes.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+- don't add new required properties to interfaces and types (only optional with '?')
+- don't add new required parameters to public functions (only optional)
+- don't change function behavior (after change it should return the same output for provided input and dispatch same side effects). Bug fixes might require different function behavior for edge cases.
+- don't change, remove unit tests (you can only change internal implementation details, assertions shouldn't be changed)
+- don't remove anything from Angular modules (imports, providers, declarations, exports)
+- check generated public API report with previous `develop` report. Make sure only things that you explicitly wanted exported are visible publicly.
+- don't remove or update existing values for public enums (only add new values)
+- default value for new functions parameters should match function behavior before change
+- any change to class constructor is breaking (`super` calls needs to be updated in classes extending ours)
+- don't change exported class fields to readonly
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md]: don't change inital values of exported class' fields or change them to readonly

reason example:
```typescript
class Pagination {
   PAGE_SIZE = 10 // default behavior of component that can a Customer may rely on
}
```

File: _pages/contributing/breaking-changes.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+- don't change or remove translations
+
+If you have any other general rule regarding breaking changes feel free to add it.
+
+Keeping enormous API we have stable is very challenging. Always think about bigger consequences of your changes and try to anticipate new problems, when changing library API.
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md]: Can we move it also to a separate paragraph about minimal API? It can be even before the Avoiding Breaking Changes.

And the whole doc could be i.e. *Maintaining public API*

File: _pages/contributing/breaking-changes.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+- don't change or remove exported classes, functions, constants, interfaces and types in `public_api.ts` file
+- don't change or remove interface properties
+- don't change or remove public/protected class members
+- don't change or remove existing public functions parameters (also don't change order)
+- don't add new required properties to interfaces and types (only optional with '?')
+- don't add new required parameters to public functions (only optional)
+- don't change function behavior (after change it should return the same output for provided input and dispatch same side effects). Bug fixes might require different function behavior for edge cases.
+- don't change, remove unit tests (you can only change internal implementation details, assertions shouldn't be changed)
+- don't remove anything from Angular modules (imports, providers, declarations, exports)
+- check generated public API report with previous `develop` report. Make sure only things that you explicitly wanted exported are visible publicly.
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md]: Can we move this to a separate paragraph `Keeping public API minimal`?
